beets/autotag/__init__.py

```diff
@@ -197,3 +211,149 @@ def apply_metadata(album_info, mapping):
             if value is None and not clobber:
                 continue
             item[field] = value
+
+
+def get_distance(config, data_source, info):
+    """Returns the ``data_source`` weight and the maximum source weight
+    for albums or individual tracks.
+    """
+    dist = Distance()
+    if info.data_source == data_source:
+        dist.add('source', config['source_weight'].as_number())
+    return dist
+
+
+@six.add_metaclass(abc.ABCMeta)
+class APIAutotaggerPlugin(object):
```

Comment:
In retrospect, perhaps the most natural home for this base class would be in the `beets.plugins` module?

In general, we have talked several times in beets's history about creating "template" base classes like this one to make it simpler to write certain kinds of plugins. So I'm very happy to see that you came to the same idea independently! It would be great to introduce more of these, so perhaps creating a new place to gather plugin templates together would be the thing to do.

Reply:
Also, what do you think about the name `MetadataSourcePlugin`?

Reply:
Awesome! I was unsure if this class is the right approach, but it's great to hear something like this was already being considered. And I agree, `beets.plugins.MetadataSourcePlugin` seems like a more natural place/name for this.